core/ecschema-editing/src/Editing/ECClasses.ts

```diff
@@ -18,40 +18,38 @@ import { assert } from "@itwin/core-bentley";
 import { PropertyEditResults, SchemaContextEditor, SchemaItemEditResults } from "./Editor";
 import { MutableClass } from "./Mutable/MutableClass";
 import * as Rules from "../Validation/ECRules";
-import { ArrayProperties, EnumerationProperties, NavigationProperties, PrimitiveProperties, Properties, StructProperties } from "./Properties";
+import { ArrayProperties, EnumerationProperties, PrimitiveProperties, Properties, StructProperties } from "./Properties";
+
+export type ECClassSchemaItems = Extract<SchemaItemType, SchemaItemType.EntityClass | SchemaItemType.StructClass | SchemaItemType.RelationshipClass | SchemaItemType.Mixin | SchemaItemType.CustomAttributeClass>;
```

Comment:
Since we do not require dynamic extracting of types here, I'd propose:
`export type ECClassSchemaItems = SchemaItemType, SchemaItemType.EntityClass | SchemaItemType.StructClass | SchemaItemType.RelationshipClass | SchemaItemType.Mixin | SchemaItemType.CustomAttributeClass;`